scripts/libraries/ml/ml/preprocessing.py

```diff
@@ -35,8 +35,18 @@ def __call__(self, *args):
 
         buffer, shape, dtype = args
         # Create an image using the input tensor as buffer.
-        pixfmt = image.GRAYSCALE if shape[3] == 1 else image.RGB565
-        img = image.Image(shape[2], shape[1], pixfmt, buffer=buffer)
+        if len(shape) == 3:
+            h, w, c = shape
+        elif len(shape) == 4:
+            b, h, w, c = shape
+            if b != 1:
+                raise ValueError("Expected 1 image per batch")
+        else:
+            raise ValueError("Expected 3D or 4D input tensor ([batch,] height, width, channels)")
```

Comment:
```suggestion
        if len(shape) != 4 or shape[0] != 1:
                raise ValueError("Expected ([batch,] height, width, channels)")
        b, h, w, c = shape
```
This for sure will always be `b, h, w, c`

Reply:
Yes, will reduce it to just sanity checks.